### PR TITLE
v2.0.0 source added.

### PR DESCRIPTION
**Pull request by DeividasBakanas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        "@types",
+        "dist",
+        "*.md",
+        "package.json",
+        "LICENSE"
+    ],
+    "bugs": {
+        "url": "https://github.com/SimplrJS/simplr-loaders/issues"
+    },
+    "homepage": "https://github.com/SimplrJS/simplr-loaders#readme",
```
Comment (GiedriusGrabauskas) [inline package.json]: Homepage without #readme

File: example/src/layout/logo.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+
+export interface Props {
+    imageClass?: string;
```
Comment (GiedriusGrabauskas) [inline example/src/layout/logo.tsx]: Rename to className

File: example/src/layout/logo.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    }
+}
+
+export class Logo extends React.Component<Props, {}> {
+    render() {
+        return <a
+            href="https://simplrjs.com"
+            className="logo-container"
+        >
+            <span className={`logo ${(this.props.imageClass != null ? this.props.imageClass : "")}`}></span>
```
Comment (GiedriusGrabauskas) [inline example/src/layout/logo.tsx]: Maybe use classnames dependency?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,240 @@
+## Ignore Visual Studio temporary files, build results, and
+## files generated by popular Visual Studio add-ons.
```
Comment (GiedriusGrabauskas) [inline .gitignore]: Minify gitignore file

File: package.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    "devDependencies": {
+        "cleanup-package-json": "^0.2.1",
+        "css-to-ts": "0.1.0-beta.9",
+        "glob-uglifyjs": "0.4.0",
```
Comment (GiedriusGrabauskas) [inline package.json]: Dont uglify ant minify files

File: src/abstractions/loader-base.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    /**
+     * Unique identifier of a loader.
+     * Used in `protected AppendStyles` to prevent styles duplications.
+     */
+    abstract LoaderId: string;
+
+    /**
+     * Puts stringified stylesheet of a loader to the `<head>`.
+     * @param styles {string} strigified stylesheet of a loader.
+     */
```
Comment (GiedriusGrabauskas) [inline src/abstractions/loader-base.ts]: Missing props param in comments

Comment (GiedriusGrabauskas) [reply src/abstractions/loader-base.ts]: Why dont use this.props?

File: src/bubble-loader/bubble-loader.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Class of bubble loader component.
+ *
+ * @export
+ * @class BubbleLoader
+ * @extends {LoaderBase<BaseProps, {}>}
+ */
+export class BubbleLoader extends LoaderBase<BaseProps, {}> {
+    constructor(props: BaseProps) {
+        super(props);
```
Comment (GiedriusGrabauskas) [inline src/bubble-loader/bubble-loader.tsx]: Missing context

File: src/abstractions/loader-base.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+ * @template TProps interface of loader component Props.
+ * @template TState interface of loader component State.
+ */
+export abstract class LoaderBase<TProps extends BaseProps, TState> extends React.PureComponent<TProps, TState> {
+
+    /**
+     * Unique identifier of a loader.
+     * Used in `protected AppendStyles` to prevent styles duplications.
+     */
+    abstract LoaderId: string;
```
Comment (GiedriusGrabauskas) [inline src/abstractions/loader-base.ts]: Wgy public?

Comment (GiedriusGrabauskas) [reply src/abstractions/loader-base.ts]: Why*

File: src/spinner-loader/spinner-loader.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Class of spinner loader component.
+ *
+ * @export
+ * @class SpinnerLoader
+ * @extends {LoaderBase<BaseProps, {}>}
+ */
+export class SpinnerLoader extends LoaderBase<BaseProps, {}> {
+    constructor(props: BaseProps) {
```
Comment (GiedriusGrabauskas) [inline src/spinner-loader/spinner-loader.tsx]: Missing context

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

File: src/abstractions/loader-base.ts [hunk 10]
```diff
@@ -65,14 +65,14 @@ export abstract class LoaderBase<TProps extends BaseProps, TState> extends React
...
+    protected abstract LoaderId: string;
 
     /**
      * Puts stringified stylesheet of a loader to the `<head>`.
      * @param styles {string} strigified stylesheet of a loader.
      */
-    protected AppendStyles(styles: string, props: TProps) {
-        if (props.useDefaultStyle === false) {
+    protected AppendStyles(styles: string) {
+        if (this.props.useDefaultStyle === false) {
```
Comment (GiedriusGrabauskas) [inline src/abstractions/loader-base.ts]: Ammm... `this.props` is not yet set.

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,6 @@
+node_modules/
```
Comment (GiedriusGrabauskas) [inline .gitignore]: Remove `/`

File: CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
+## Changes in 2.0.0 (2017-04-18)
+
+### [Breaking changes]
+- Directories structure changed. All production files moved to `dist` folder.
+- `blackLoader` prop deprecated in `SpinnerLoader`.
+- `color` prop deprecated in `BubbleLoader`.
```
Comment (GiedriusGrabauskas) [inline CHANGELOG.md]: Update decription to `removed`

File: README.md [hunk 13]
```diff
@@ -1,2 +1,181 @@
...
+
+# Usage
+
+```ts
+import * as React from "react";
+import { BubbleLoader } from "simplr-loaders";
+
+export class MyComponent extends React.Component<{}, {}> {
+    render() {
+        return <BubbleLoader/>;
```
Comment (GiedriusGrabauskas) [inline README.md]: Spacing.
```ts
return <BubbleLoader />;
```

File: example/package.json [hunk 14]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "simplr-loaders-example",
+  "version": "0.1.0",
+  "description": "Example for simplr-loaders.",
```
Comment (GiedriusGrabauskas) [inline example/package.json:4]: Add `private: true`

File: package.json [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+    "build": "npm run clean && npm run build-scss && npm run css-to-ts && npm run build-ts",
+    "watch": "tsc -w & npm run css-to-ts -- -w & npm run builc-scss -- -w",
+    "clean": "rm -r -f @types & rm -r -f dist",
+    "css-to-ts": "css-to-ts --rootDir \"./dist\" --outDir \"./src\" --suffix style --header \"tslint:disable-next-line:max-line-length\" --exclude **/node_modules/** example/**",
+    "build-scss": "node-sass --output-style compressed ./src -o ./dist",
+    "build-ts": "tsc -p .",
+    "tslint": "tslint --project \"./tsconfig.json\" --c \"./tslint.json\"",
+    "uglifyjs": "glob-uglifyjs",
+    "release": "npm run build && npm run tslint && npm run uglifyjs",
+    "cleanup-package-json": "cpj"
```
Comment (GiedriusGrabauskas) [inline package.json]: Remove cpj

File: package.json [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+    "css-to-ts": "0.1.0-beta.9",
+    "glob-uglifyjs": "0.4.0",
+    "node-sass": "^4.5.0",
+    "pushstate-server": "^3.0.0",
+    "tslint": "^5.1.0",
+    "typescript": "^2.0.9"
+  },
+  "dependencies": {
+    "react": "^15.3.2",
+    "react-dom": "^15.3.2",
```
Comment (GiedriusGrabauskas) [inline package.json]: react and react-dom move to peerDependencies

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

**Review by DovydasNavickas — APPROVED** [review]

File: @types/abstractions/loader-base.d.ts [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Specifies whether style should be placed in <head>.
+     * This is a default behavior you can cancel.
+     *
+     * @type {boolean}
+     * @memberOf BaseProps
+     */
+    useDefaultStyle?: boolean;
+    /**
+     * Specifies whether loader should take area of a strict size.
```
Comment (DovydasNavickas) [inline @types/abstractions/loader-base.d.ts]: Specifies whether the loader should take an area of a strict size.

File: @types/abstractions/loader-base.d.ts [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+     * By default a loader should expand to take all possible area.
+     * Using this prop you can cancel default behavior.
```
Comment (DovydasNavickas) [inline @types/abstractions/loader-base.d.ts]: the default behaviour

File: @types/abstractions/loader-base.d.ts [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+     *
+     * @type {boolean}
+     * @memberOf BaseProps
+     */
+    shouldReduceSize?: boolean;
```
Comment (DovydasNavickas) [inline @types/abstractions/loader-base.d.ts]: shouldReduceSize -> shouldExpand

File: README.md [hunk 20]
```diff
@@ -1,2 +1,181 @@
-# simplr-loaders
-React css loaders with no prerequisite style loading.
+# simplr-loaders
+React css loaders with no prerequisite style loading.
```
Comment (DovydasNavickas) [inline README.md]: React CSS loaders with no prerequisites for required styles loading.

**Review by DovydasNavickas — CHANGES_REQUESTED** [review]

**Review by DovydasNavickas — APPROVED** [review]

**Review by GiedriusGrabauskas — APPROVED** [review]

**Pull request merged**